pkg/helm/charts.go

```diff
@@ -302,6 +302,44 @@ func (c *Charts) AddRepos(repos ...Repo) error {
 	return nil
 }
 
+// VersionCheck checks each of the charts in the requires section and returns information regarding
+// related to version upgrades. This includes if the current version is latest as well as the
+// latest matching versions of the major and minor version the chart is currently on.
+func (c *Charts) VersionCheck(repoConfigPath string) (map[string]RequiresVersionInfo, error) {
+	requiresVersionInfo := make(map[string]RequiresVersionInfo)
+
+	if repoConfigPath != "" {
+		repoConfig, err := LoadHelmRepoConfig(repoConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		c.Manifest.Repositories = repoConfig.Repositories
+	}
```

Comment:
I think this should be a small private func of *Charts. Also, I just noticed, but I'm not a fan of mutating the charts object. I think you should return another var and use that instead

```suggestion
	repositories := c.GetRepositories(repoConfigPath)
```

```
	func (c *Charts) GetRepositories(repoConfigPath string) Repos {
             if repoConfigPath != "" {
		repoConfig, err := LoadHelmRepoConfig(repoConfigPath)
		if err != nil {
			return nil, err
		}
		return repoConfig.Repositories
	}
        return c.Manifest.Repositories
}
```